synapseclient/client.py

```diff
@@ -721,16 +721,22 @@ def get(self, entity, **kwargs):
             version = kwargs.get('version', None)
             bundle = self._getEntityBundle(entity, version)
         # Check and warn for unmet access requirements
-        self._check_entity_restrictions(bundle['restrictionInformation'], entity, kwargs.get('downloadFile', True))
+        # self._check_entity_restrictions(bundle['restrictionInformation'], entity, kwargs.get('downloadFile', True))
+        self._check_entity_restrictions(bundle, kwargs.get('downloadFile', True))
 
         return self._getWithEntityBundle(entityBundle=bundle, entity=entity, **kwargs)
 
-    def _check_entity_restrictions(self, restrictionInformation, entity, downloadFile):
+    # def _check_entity_restrictions(self, restrictionInformation, entity, downloadFile):
+    def _check_entity_restrictions(self, bundle, downloadFile):
+        restrictionInformation = bundle['restrictionInformation']
         if restrictionInformation['hasUnmetAccessRequirement']:
             warning_message = ("\nThis entity has access restrictions. Please visit the web page for this entity "
                                "(syn.onweb(\"%s\")). Click the downward pointing arrow next to the file's name to "
-                               "review and fulfill its download requirement(s).\n" % id_of(entity))
-            if downloadFile:
+                               "review and fulfill its download requirement(s).\n" % id_of(bundle['entity']['id']))
+            # warning_message = ("\nThis entity has access restrictions. Please visit the web page for this entity "
+            #                    "(syn.onweb(\"%s\")). Click the downward pointing arrow next to the file's name to "
+            #                    "review and fulfill its download requirement(s).\n" % id_of(entity))
+            if downloadFile and bundle['entityType'] not in ["project", "folder"]:
```

Comment:
maybe
`if downloadFile and bundle.get('entityType') not in ('project', 'folder'):`

get is a bit safer in cases where we are getting something from the backend in case the object doesn't have an expected key, and I tend to prefer tuples rather than lists for checking collection literals like this (since they are immutable etc).